src/sdk.rs

```diff
@@ -157,20 +162,28 @@ mod exports {
 
 pub fn read_input() -> Vec<u8> {
     unsafe {
-        exports::input(0);
-        let bytes: Vec<u8> = vec![0; exports::register_len(0) as usize];
-        exports::read_register(0, bytes.as_ptr() as *const u64 as u64);
+        exports::input(INPUT_REGISTER_ID);
```

Comment:
@birchmd IMHO all those changes really redundant.
WASM has an infinity register stack. We don't need to unify as some standard `get/read/write` register flow. Most important read from the same register. And it was. it will be misleading that we should only use one given register.

Reply:
I think giving constants a name is useful for two reasons:

1. It provides a semantic meaning to an otherwise ambiguous value (`0` could mean lots of things, but here it is a register ID)
2. It makes changing the value less error prone in the future because all occurrences of it now point back to one definition.  For example, if we decide to reserve register 0 for some purpose then we can change `INPUT_REGISTER_ID` to be some other value easily without carefully double-checking we changed all the right places.